src/objectid.ts

```diff
@@ -272,18 +342,18 @@ export class ObjectId extends BSONValue {
 
   /** @internal */
   serializeInto(uint8array: Uint8Array, index: number): 12 {
-    uint8array[index] = this.buffer[0];
-    uint8array[index + 1] = this.buffer[1];
-    uint8array[index + 2] = this.buffer[2];
-    uint8array[index + 3] = this.buffer[3];
-    uint8array[index + 4] = this.buffer[4];
-    uint8array[index + 5] = this.buffer[5];
-    uint8array[index + 6] = this.buffer[6];
-    uint8array[index + 7] = this.buffer[7];
-    uint8array[index + 8] = this.buffer[8];
-    uint8array[index + 9] = this.buffer[9];
-    uint8array[index + 10] = this.buffer[10];
-    uint8array[index + 11] = this.buffer[11];
+    uint8array[index] = this[_pool][this[_offset]];
+    uint8array[index + 1] = this[_pool][this[_offset] + 1];
+    uint8array[index + 2] = this[_pool][this[_offset] + 2];
+    uint8array[index + 3] = this[_pool][this[_offset] + 3];
+    uint8array[index + 4] = this[_pool][this[_offset] + 4];
+    uint8array[index + 5] = this[_pool][this[_offset] + 5];
+    uint8array[index + 6] = this[_pool][this[_offset] + 6];
+    uint8array[index + 7] = this[_pool][this[_offset] + 7];
+    uint8array[index + 8] = this[_pool][this[_offset] + 8];
+    uint8array[index + 9] = this[_pool][this[_offset] + 9];
+    uint8array[index + 10] = this[_pool][this[_offset] + 10];
+    uint8array[index + 11] = this[_pool][this[_offset] + 11];
```

Comment:
Not sure if it really makes a difference, but my intuition would be that it's worth storing `this[_pool]` and `this[_offset]` in separate variables?

Reply:
Just so I understand correctly, do you mean using some temp variables like this? Would make it easier to grok for sure.

```ts
serializeInto(uint8array: Uint8Array, index: number): 12 {
  const pool = this[_pool]
  const offset = this[_offset]
   ...
  uint8array[index + 1] = pool[offset + 1];
  ...
```

Reply:
Yes, exactly 👍

Reply:
Even if we switch to the string properties I think this is a good call

> Would make it easier to grok for sure.

Aggreed, more so for how it reads rather than any functional difference.